tortoise/contrib/pydantic/creator.py

```diff
@@ -155,332 +177,403 @@ def pydantic_model_creator(
-        __validators__=validators,
-        **properties,
-    )
-    # Copy the Model docstring over
-    model.__doc__ = _cleandoc(cls)
-    # Store the base class
-    model.model_config["orig_model"] = cls  # type: ignore
-    # Store model reference so we can de-dup it later on if needed.
-    _MODEL_INDEX[_name] = model
-    return model
+    def _process_data_field(
+            self,
+            field_name: str,
+            field: Field,
+            json_schema_extra: Dict[str, Any],
+            fconfig: Dict[str, Any],
+    ) -> Optional[Any]:
+        annotation = self._annotations.get(field_name, None)
+        constraints = copy(field.constraints)
+        if "readOnly" in constraints:
+            json_schema_extra["readOnly"] = constraints["readOnly"]
+            del constraints["readOnly"]
+        fconfig.update(constraints)
+        python_type = getattr(field, "related_model", field.field_type)
+        ptype = python_type
+        if field.null:
+            json_schema_extra["nullable"] = True
+        if field_name in self._optional or field.default is not None or field.null:
+            ptype = Optional[ptype]
+        if not (self._exclude_read_only and json_schema_extra.get("readOnly") is True):
+            return annotation or ptype
+        return None
 
+    def _process_computed_field(
+            self,
+            field: ComputedFieldDescription,
+    ) -> Optional[Any]:
+        func = field.function
+        annotation = get_annotations(self._cls, func).get("return", None)
+        comment = _cleandoc(func)
+        if annotation is not None:
+            c_f = computed_field(return_type=annotation, description=comment)
+            ret = c_f(func)
+            return ret
+        return None
 
-def pydantic_queryset_creator(
-    cls: "Type[Model]",
-    *,
-    name=None,
-    exclude: Tuple[str, ...] = (),
-    include: Tuple[str, ...] = (),
-    computed: Tuple[str, ...] = (),
-    allow_cycles: Optional[bool] = None,
-    sort_alphabetically: Optional[bool] = None,
-) -> Type[PydanticListModel]:
+    def _get_submodel(self, _model: Optional["Type[Model]"], field_name: str) -> Optional[Type[PydanticModel]]:
+        """Get Pydantic model for the submodel"""
+
+        if _model:
+            new_stack = self._stack + ((self._cls, field_name, self.meta.max_recursion),)
+
+            # Get pydantic schema for the submodel
+            prefix_len = len(field_name) + 1
+            pmodel = _pydantic_recursion_protector(
+                _model,
+                exclude=tuple(
+                    str(v[prefix_len:]) for v in self.meta.exclude if v.startswith(field_name + ".")
+                ),
+                include=tuple(
+                    str(v[prefix_len:]) for v in self.meta.include if v.startswith(field_name + ".")
+                ),
+                computed=tuple(
+                    str(v[prefix_len:]) for v in self.meta.computed if v.startswith(field_name + ".")
+                ),
```

Comment:
May be we can make this transformation into a function? Inline function would be good enough

Reply:
I did not want to use a lambda expression here (pycharm yelled at me "PEP 8: E731 do not assign a lambda expression, use a def"), so I put it in an inner function